pyanaconda/ui/lib/help.py

```diff
@@ -19,133 +19,165 @@
-
-    # setup the fallback files
     if plain_text:
-        main_page = HELP_MAIN_PAGE_TUI
-        placeholder = conf.ui.default_help_pages[0]
-    elif not conf.system.provides_web_browser:
-        main_page = HELP_MAIN_PAGE_GUI
-        placeholder = conf.ui.default_help_pages[1]
+        placeholder = conf.ui.help_placeholder_plain_text
+    else:
+        placeholder = conf.ui.help_placeholder
+    placeholder_path = get_best_help_file(conf.ui.help_directory, placeholder)
+    if placeholder_path:
+        return placeholder_path
     else:
-        main_page = HELP_MAIN_PAGE_GUI
-        placeholder = conf.ui.default_help_pages[2]
+        log.error("placeholder %s not found in %s",
+                  conf.ui.help_placeholder,
+                  conf.ui.help_directory)
+        return None
 
-    # the screen did not have a helpFile defined or the defined help file
-    # does not exist, so next try to check if we can find the main page
-    # of the installation guide and use it instead
-    help_path = _get_best_help_file(main_page)
-    if help_path is not None:
-        return help_path
 
-    # looks like the installation guide is not available, so just return
-    # a placeholder page, which should be always present
-    return _get_best_help_file(placeholder)
+def start_yelp(help_id, help_directory=conf.ui.help_directory):
+    """Start a new yelp process and make sure to kill any existing ones.
 
+    We will try to resolve the help id to appropriate content and show it to the user.
 
-def start_yelp(help_path):
+    :param str help_id: help id to be shown to the user
+    :param str help_directory: path to directory containing help content
     """
-    Start a new yelp process and make sure to kill any existing ones
+    kill_yelp()
 
-    :param help_path: path to the help file yelp should load
-    :type help_path: str or NoneType
-    """
+    log.info("help requested for help id %s", help_id)
 
-    kill_yelp()
-    log.debug("starting yelp")
-    global yelp_process
-    # under some extreme circumstances (placeholders missing)
-    # the help path can be None and we need to prevent Popen
-    # receiving None as an argument instead of a string
-    yelp_process = startProgram(["yelp", help_path or ""], reset_lang=False)
+    # get the help id -> mapping
+    help_id_mapping = get_help_id_to_anchor_mapping(help_directory)
+
+    # we initially set the yelp string to the placeholder path, just in case
+    yelp_string = get_placeholder_path()
+
+    # resolve the help id to docbook document anchor (if there is one for the help id)
+    help_anchor = help_id_mapping.get(help_id, None)
+    if help_anchor is None:
+        log.error("no anchor found for help id %s, starting yelp with placeholder", help_id)
+    else:
+        log.info("help id %s was resolved to anchor %s", help_id, help_anchor)
+        # check if main help file is available
+        main_help_file_path = get_best_help_file(help_directory, MAIN_HELP_FILE)
```

Comment:
You can the path at the same time as the anchor. Then the code can be simplified. If you have a path and an anchor, use them to generate the yelp string, otherwise use a default file as the yelp string.

Reply:
Sounds good, lets do that. :)